server/src/main/java/org/cloudfoundry/identity/uaa/web/beans/UaaDefaultCookieSerializer.java

```diff
@@ -0,0 +1,434 @@
+     * can help with tracing logs of a particular user. This will ensure that the value of
+     * the cookie is formatted as
+     * </p>
+     * <code>
+     * sessionId + "." jvmRoute
+     * </code>
+     * <p>
+     * To use set a custom route on each JVM instance and setup a frontend proxy to
+     * forward all requests to the JVM based on the route.
+     * </p>
+     *
+     * @param jvmRoute the JVM Route to use (i.e. "node01jvmA", "n01ja", etc)
+     */
+    public void setJvmRoute(String jvmRoute) {
+        this.jvmRoute = "." + jvmRoute;
+    }
+
+    /**
+     * Set if the Base64 encoding of cookie value should be used. This is valuable in
+     * order to support <a href="https://tools.ietf.org/html/rfc6265">RFC 6265</a> which
+     * recommends using Base 64 encoding to the cookie value.
+     *
+     * @param useBase64Encoding the flag to indicate whether to use Base64 encoding
+     */
+    public void setUseBase64Encoding(boolean useBase64Encoding) {
+        this.useBase64Encoding = useBase64Encoding;
+    }
+
+    /**
+     * Set the request attribute name that indicates remember-me login. If specified, the
+     * cookie will be written as Integer.MAX_VALUE.
+     * @param rememberMeRequestAttribute the remember-me request attribute name
+     * @since 1.3.0
+     */
+    public void setRememberMeRequestAttribute(String rememberMeRequestAttribute) {
+        if (rememberMeRequestAttribute == null) {
+            throw new IllegalArgumentException(
+                "rememberMeRequestAttribute cannot be null");
+        }
+        this.rememberMeRequestAttribute = rememberMeRequestAttribute;
+    }
+
+    /**
+     * Set the value for the {@code SameSite} cookie directive. The default value is
+     * {@code Lax}.
+     * @param sameSite the SameSite directive value
+     * @since 2.1.0
+     */
+    public void setSameSite(String sameSite) {
+        this.sameSite = sameSite;
+    }
+
+    private String getDomainName(HttpServletRequest request) {
+        if (this.domainName != null) {
+            return this.domainName;
+        }
+        if (this.domainNamePattern != null) {
+            Matcher matcher = this.domainNamePattern.matcher(request.getServerName());
+            if (matcher.matches()) {
+                return matcher.group(1);
+            }
+        }
+        return null;
+    }
+
+    private String getCookiePath(HttpServletRequest request) {
+        if (this.cookiePath == null) {
+            return request.getContextPath() + "/";
+        }
+        return this.cookiePath;
+    }
+
+    /**
+     * Returns true if the Servlet 3 APIs are detected.
+     *
+     * @return whether the Servlet 3 APIs are detected
+     */
+    private boolean isServlet3() {
+        try {
+            ServletRequest.class.getMethod("startAsync");
+            return true;
+        }
+        catch (NoSuchMethodException e) {
+        }
+        return false;
+    }
+
+}
```

Comment:
Minor issue, add a newline at the end of the file.

Reply:
Done